src/interrupt/trigger/mod.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2021 Amazon.com, Inc. or its affiliates.
+// All Rights Reserved.
+
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+//! Triggers that are commonly used by multiple VMMs.
+
+#[cfg(target_os = "linux")]
+pub mod eventfd;
+
+/// Abstraction for a simple, push-button like interrupt mechanism.
+/// This helps in abstracting away how events/interrupts are generated when
+/// working with the emulated devices.
+///
+/// The user has to provide a `Trigger` object to the device's constructor when
+/// initializing that device. The generic type `T: Trigger` is used in the
+/// device's structure definition to mark the fact that the events notification
+/// mechanism is done via the Trigger interface.
+pub trait Trigger {
+    /// Underlying type for the potential error conditions returned by `Self::trigger`.
+    type E: std::fmt::Debug;
```

Comment:
Why the `Debug` trait bound? If we want to communicate that `E` is an error, then the bound should be `std::error::Error`, unless there are good reasons not to. Alternatively we could drop the bound altogether and just leave the associated type as `type E;`.

Reply:
It's actually the definition moved from here: https://github.com/rust-vmm/vm-superio/blob/14906a0e5363557c9c97632b03f986135c4f5d3b/crates/vm-superio/src/lib.rs#L47

It actually makes more sense to bind it to `std::error::Error` though users of this trait from vm-superio might need to change their implementations. But I guess that maybe it's best to change it now rather than later.

Reply:
Can we instead merge this PR: https://github.com/rust-vmm/vm-device/pull/29/ and then rebase this one on top of it? We can do the adjustments in vm-superio afterwards as right now vm-superio defines its own trait anyway.

Reply:
Actually I've imported the commit from #29 an rebased it on top of main. It is now part of this PR.

Reply:
Can you also add Alex's signed-off line to the commit message? We are trying to keep the original author of the commits that we cherry-pick. Example [here](https://github.com/rust-vmm/vm-superio/commit/7f3cb2b6a3f42b4c5c1faf414e8bd35030ab4f9b).